code/android-core-library/src/test/java/com/adobe/marketing/mobile/internal/configuration/MobileIdentitiesProviderTest.kt

```diff
@@ -0,0 +1,313 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.internal.configuration
+
+import com.adobe.marketing.mobile.Event
+import com.adobe.marketing.mobile.ExtensionApi
+import com.adobe.marketing.mobile.SharedStateResolution
+import com.adobe.marketing.mobile.SharedStateResult
+import com.adobe.marketing.mobile.SharedStateStatus
+import com.adobe.marketing.mobile.VisitorID
+import com.adobe.marketing.mobile.internal.util.VisitorIDSerializer
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.anyString
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.eq
+import kotlin.test.assertEquals
+
+class MobileIdentitiesProviderTest {
+    @Mock
+    private lateinit var mockExtensionApi: ExtensionApi
```

Comment:
Are these the same tests from our swift implementation?

Reply:
Yea, these include all tests from swift and a few additional tests.